### ISSUE TITLE
Lever compatibility?

### ISSUE THREAD

**Issue by Larsg310** [amadornes/MCMultiPart#54]
Is there a way to make levers and torches compatible with MCMultiPart? For example, make Simple Covers covers be able to be placed in the same block space as a lever, or Chisel and Bits bits?

That would be much appreciated.

**Comment by amadornes** [comment]
Yes, that is planned. However, it won't be part of MCMultiPart, it'll be in MCMicroBlock, the official microblock mod for MCMultiPart. MCMP is only meant to let people add multiparts, not alter the game, which is why it's not going to be in natively.

**Comment by Dushess** [comment]
Will ask also about signboards, it can be converted to MCMP compatibility too? Maybe even without SimpleCovers support, it can be handled by C&B bits.
In addition, buttons, i remember they was Multipart compatible too in 1.7.10

**Comment by amadornes** [comment]
I'd suggest waiting until MCMP 2.0.0 drops (in 1.11) for any kind of vanilla compat or mod compatibility in general. I'm rewriting the API so it's extremely easy to turn any already existing block into a multipart with little to no effort (like seriously, it's just adding an interface to your block class or registering a wrapper through the API, it takes less than 20 lines).
If the move to 1.11 takes a while (which I hope isn't the case), I may look into writing some kind of wrapper for already existing blocks and TEs and maybe encouraging more people to use MCMP, but the new API is far superior and it probably makes everyone who doesn't currently support it want to hold off and not waste time and effort.

**Comment by Dushess** [comment]
We should wait it for currently used 1.10.2 sometime?

I also think about SignPicture compatibility, it uses sign for attaching an entity, that containing image. If make it part of MCMP, but there's some problems.
 [Here](https://github.com/Team-Fruit/SignPicture/issues/5#issuecomment-261995625) we talking about this.

**Comment by sjcl** [comment]
SignPicture works only if the character string written in Sign can be read. 
In most cases, it will not be impossible.

**Comment by Dushess** [comment]
@sjcl 
Yes, because sign don't loses his functional, it just becomes part of MCMP, i think not all is lost.

**Comment by amadornes** [comment]
Yeah, everything inside an MCMultiPart block is still fully functional. The problem is adapting already existing blocks (vanilla signs in this case) to work as multiparts. In 1.10 MCMP it's not very easy, but in 1.11 it'll literally be about 2 lines of code per block if I write a generic wrapper, and I'll be adding support myself for all vanilla blocks, so you shouldn't even need to touch the API :)
